fearless/Modules/Staking/Analytics/Model/AnalyticsPeriod.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+
+enum AnalyticsPeriod: CaseIterable {
+    case weekly
+    case monthly
+    case yearly
+}
+
+extension AnalyticsPeriod {
+    static let `default` = AnalyticsPeriod.monthly
+
+    func title(for _: Locale) -> String {
+        switch self {
+        case .weekly:
+            return "weekly".uppercased()
+        case .monthly:
+            return "monthly".uppercased()
+        case .yearly:
+            return "yearly".uppercased()
+        }
+    }
+
+    var chartBarsCount: Int {
+        xAxisValues.count
+    }
+
+    var xAxisValues: [String] {
+        switch self {
+        case .weekly:
+            return ["M", "T", "W", "T", "F", "S", "S"]
+        case .monthly:
+            return ["1", "7", "15", "22", "31"]
+        case .yearly:
+            return (1 ... 12).map { String($0) }
```

Comment:
I think we should be consistent with a calendar here? We use `Calendar.current` for grouping rewards. But no all calendars has 12 months in a year, I think)